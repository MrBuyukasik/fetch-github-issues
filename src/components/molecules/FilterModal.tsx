import React from 'react';
import ModalDropdown from 'react-native-modal-dropdown';

interface Props {
  selectedFilter: Function;
}

export const issueStatus = ['open', 'closed', 'all'];

const FilterModal = (props: Props) => {
  const defaultvValue = 'Filter Issues';
  const style = {borderWidth: 1, borderColor: 'black', padding: 5};
  const dropdownTextStyle = {fontSize: 14};
  const textStyle = {fontSize: 12};
  const dropdownStyle = {width: '30%'};

  return (
    <ModalDropdown
      style={style}
      dropdownTextStyle={dropdownTextStyle}
      textStyle={textStyle}
      dropdownStyle={dropdownStyle}
      options={issueStatus}
      defaultValue={defaultvValue}
      onSelect={value => {
        props.selectedFilter(issueStatus[value]);
      }}
    />
  );
};

export default FilterModal;
