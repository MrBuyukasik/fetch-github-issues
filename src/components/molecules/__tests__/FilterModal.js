import React from 'react';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import FilterModal from '../FilterModal';

describe('Filter Modal test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FilterModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
