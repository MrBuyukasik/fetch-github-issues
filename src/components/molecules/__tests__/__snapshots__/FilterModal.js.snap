// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Modal test renders correctly 1`] = `
<View
  animated={true}
  defaultIndex={-1}
  defaultValue="Filter Issues"
  disabled={false}
  dropdownStyle={
    Object {
      "width": "30%",
    }
  }
  dropdownTextStyle={
    Object {
      "fontSize": 14,
    }
  }
  isFullWidth={false}
  keySearchObject="label"
  keyboardShouldPersistTaps="never"
  multipleSelect={false}
  numberOfLines={1}
  onSelect={[Function]}
  options={
    Array [
      "open",
      "closed",
      "all",
    ]
  }
  renderButtonComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  renderRightComponent={[Function]}
  renderRowComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  saveScrollPosition={true}
  scrollEnabled={true}
  searchPlaceholder="Search"
  showSearch={false}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "borderColor": "black",
      "borderWidth": 1,
      "padding": 5,
    }
  }
  textStyle={
    Object {
      "fontSize": 12,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Array [
              Object {
                "fontSize": 12,
              },
              undefined,
            ],
          ]
        }
      >
        Filter Issues
      </Text>
      <View />
    </View>
  </View>
</View>
`;
